README.md

```diff
@@ -8,27 +8,194 @@
+
+### Body Mixin
+
+The `fetch` API includes a *mixin*, or additional code, called
+[Body](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Body), which has functions that specialize in transforming the `body` of a request or response.
+
+In an `XMLHttpRequest`, you would potentially have to `JSON.parse` the
+`responseText` data to turn it into JSON. The `Body` mixin takes that a
+step further, giving us handy ways to parse many formats, including
+JSON, plain text, and even blobs for binary or image data.
+
+So calling `res.json()` in our `fetch` is just a nice shorthand for
+saying "give me the `body` of the response parsed as JSON".
+
+## Authenticated Fetching
+
+So far, we've been making our requests against the public API functions
+that Github provides, but there are more powerful toys available to us
+if we can prove to the API who we are when we make requests.
+
+### OAuth
+
+Github's API uses [OAuth2](https://developer.github.com/v3/oauth/) for
+authorization. In a production setting, you would register an
+application with Github and receive an application ID and secret. This
+allows Github to track and monitor API access, and ensure that its users
+are protected by only granting authorization through registered apps.
+
+From there, a user would make a request via your app, which would kick
+off the workflow to issue an authorization code that your app would then
+exchange for an access token. The access token is ultimately what your
+application includes in Github API requests to identify and authorize
+the user.
+
+Setting up the full OAuth2 authorization code grant workflow is beyond
+the scope of this lesson, but it is described well in the Github
+[docs](https://developer.github.com/v3/oauth/).
 
-If there's time (after the intro, getting them into the promise-like syntax, etc.), it might be worth having them sign up for a service and pass tokens in the `headers` part of the options, e.g.,
+Fortunately for us, Github also allows you to generate your own personal
+authorization token that we can use to give us authorized access to the
+API. And `fetch` makes it super easy to implement.
 
-```javascript
-fetch(`https://api.github.com/users/${username}`, { 
-  headers: { 
-    Authorization: 'token ${token}'
+To start, go to https://github.com/settings/tokens and click "Generate
+new token." Name it "Learn.co" and check `repo` scope. Once you generate
+the token, make sure to copy and paste it somewhere, because once you
+leave that page, you won't be able to see it again. This is for your
+security - even if someone were to gain access to this page on your
+account, they still couldn't see your tokens.
+
+Using the token to [access the API](https://developer.github.com/v3/oauth/#3-use-the-access-token-to-access-the-api) is a simple matter of creating an `Authorization` header with our request. Let's try it out by listing our repos.
+
+```js
+fetch("https://api.github.com/user/repos").
```

Comment:
Awesome -- stuff like this makes students less afraid of errors and more willing to experiment.